SDK/AppCenter/Microsoft.AppCenter.WindowsDesktop/Utils/ApplicationLifecycleHelper.cs

```diff
@@ -122,10 +122,23 @@ private static bool IsAnyWindowNotMinimized()
         public ApplicationLifecycleHelper()
         {
             Enabled = true;
-            AppDomain.CurrentDomain.UnhandledException += (sender, eventArgs) =>
+
+            // Add the event handler for handling UI thread exceptions to the event.
+            Application.ThreadException += (sender, args) => InvokeUnhandledException(sender, args.Exception);
+
+            // Set the unhandled exception mode to force all Windows Forms errors to go through our handler.
+            Application.SetUnhandledExceptionMode(UnhandledExceptionMode.CatchException);
+
+            // Add the event handler for handling non-UI thread exceptions to the event. 
+            AppDomain.CurrentDomain.UnhandledException += (sender, args) => InvokeUnhandledException(sender, args.ExceptionObject as Exception);
+        }
+
+        private void InvokeUnhandledException(object sender, Exception exception)
+        {
+            if (exception != null)
             {
-                UnhandledExceptionOccurred?.Invoke(sender, new UnhandledExceptionOccurredEventArgs((Exception)eventArgs.ExceptionObject));
-            };
+                UnhandledExceptionOccurred?.Invoke(sender, new UnhandledExceptionOccurredEventArgs(exception));
```

Comment:
should we check for double invocation now? what does hockey app do?

Reply:
I don't think so, since HA doesn't + it's invoked by different conditions

Reply:
Fair enough.